test/test_autoextract.py

```diff
@@ -0,0 +1,110 @@
+import os
+import sys
+import pytest
+
+from w3lib.http import basic_auth_header
+from scrapy.http import Request, Response
+from scrapy.spiders import Spider
+from scrapy.utils.test import get_crawler
+
+sys.path.append(os.getcwd())
+from scrapy_autoextract import AutoExtractMiddleware
+from scrapy_autoextract.middlewares import AutoExtractConfigError
+
+AUTOX_META = {'autoextract': {'enabled': True}}
+
+MW_SETTINGS = {
+    'AUTOEXTRACT_USER': 'apikey',
+    'AUTOEXTRACT_PAGE_TYPE': 'article',
+}
+
+
+def setup_module(module):
+    global spider
+    spider = Spider('spidr')
+
+
+def _mock_mw(spider, settings=None):
+
+    class MockedDownloader:
+        slots = {}
+
+        def _get_slot_key(self, a, b):
+            return str(a) + str(b)
+
+    class MockedEngine:
+        downloader = MockedDownloader()
+        fake_spider_closed_result = None
+
+        def close_spider(self, spider, reason):
+            self.fake_spider_closed_result = (spider, reason)
+
+    # with `spider` instead of `type(spider)` raises an exception
+    crawler = get_crawler(type(spider), settings)
+    crawler.engine = MockedEngine()
+    return AutoExtractMiddleware.from_crawler(crawler)
+
+
+def _assert_disabled(spider, settings=None):
+    mw = _mock_mw(spider, settings)
+    req = Request('http://quotes.toscrape.com', meta=AUTOX_META)
+    out = mw.process_request(req, spider)
+    assert out is None
+    assert req.meta.get('autoextract') is None
+    res = Response(req.url, request=req)
+    assert mw.process_response(req, res, spider) == res
+
+
+def _assert_enabled(spider,
+                    settings=None,
+                    url='http://quotes.toscrape.com',
+                    proxyurl='autoextract.scrapinghub.com',
```

Comment:
probably makes sense to rename "proxy..." arguments to something else, as AutoExtract is not a proxy